TheMoviesDB/app/src/main/java/com/ekremsenturk/themoviesdb/ui/detail/DetailViewModel.kt

```diff
@@ -0,0 +1,53 @@
+package com.ekremsenturk.themoviesdb.ui.detail
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.ekremsenturk.core.MovieRepository
+import kotlinx.coroutines.launch
+
+class DetailViewModel(
+    private val movieRepository: MovieRepository
+) : ViewModel() {
+
+    var detailUiState by mutableStateOf(DetailUiState(
+        loading = true,
+        title = "",
+        backdrop = null,
+        poster = null,
+        overview = "",
+        releaseDate = "",
+        genres = "",
+        voteCount = 0,
+        voteAverage = 0.0f,
+        cast = emptyList(),
+        crew = emptyList(),
+        recommendations = emptyList(),
+        similarMovies = emptyList()
+    ))
+        private set
+
+    fun onStart(movieId: Int) {
+        viewModelScope.launch {
```

Comment:
How could we avoid triggering the coroutine again, if onStart() would be called by mistake several times?